### DISCUSSION TITLE
Do we need to redo scv.pp.filter_and_normalize() if using the Scanpy processed data?

### DISCUSSION THREAD

**Discussion by hyjforesight** [theislab/scvelo#755] [Q&A]
Hello scVelo,
I use Scanpy to process the data (already normalized, HVG, scaled, PCAed...UMAP). And then I do scVelo on this processed data.
However, I found that that scVelo tutorial mentioned that "Filtering and normalization is applied in the same vein to spliced/unspliced counts and X." It means scVelo needs the normalized spliced/unspliced data.
I'm wondering whether Scanpy did this normalization for spliced/unspliced data and do we need to redo scv.pp.filter_and_normalize() on the Scanpy processed data?
Thanks!
Best,
YJ

**Comment by WeilerP** [comment]
Scanpy does not preprocess the unspliced/spliced counts as they are layers. You can also not filter based on shared counts in the two modalities, for example. If you've already subsetted to HVG, you'll have to redo the entire preprocessing, otherwise you can check if  `filter_and_normalize` gives the expected output. Not sure ATM if everything works as expected when calling `filter_and_normalize` on preprocessed data, *i.e.* if `.X` is left unchanged, etc.
